content/api/commands/attachfile.md

```diff
@@ -0,0 +1,213 @@
+- `drag-n-drop`
+Setting the action to `drag-n-drop` changes the behavior of the command to instead mimic a user dragging files from the operating system into the browser, and dropping them over the selected subject. In this mode, the subject can be any DOM element, or the `document` as a whole.
+
+## Examples
+
+### From a file on disk
+
+```javascript
+cy.get('input[type=file]').attachFile('file.json')
+cy.get('input[type=file]').attachFile('file.png')
+```
+
+If given a path, `cy.readFile()` will attach the file exactly as it exists on disk. This is the preferred way to work with files on disk, avoiding many encoding-related pitfalls.
+
+### On a hidden input
+
+```javascript
+cy.get('input[type=file]').attachFile('file.json', { force: true })
+```
+
+In many cases in modern applications, the underlaying file input is hidden from view, and activated by a user clicking on a button. In these cases, you will need to tell Cypress to ignore its actionability checks and attach the file even though a user would not be able to directly activate the file input.
+
+### From a fixture
+```javascript
+cy.fixture('file.json', { encoding: null }).as('myFixture')
+cy.get('input[type=file]').attachFile('@myFixture')
+```
+
+Note the use of `null` encoding. By default, `cy.fixture()` `cy.readFile()` attempt to interpret files read from disk, which would result in a json file being decoded and re-encoded as a utf-8 string - the contents would be preserved, but formatting would not be, and the encoding might change. See [`cy.fixture`](/api/commands/fixture) or [`cy.readFile`](/api/commands/readfile) for more details on file encoding.
+
+### From an API response
+
+```javascript
+cy.request('http://localhost:8888/users/827').get('body').as('responseBody')
+cy.get('input[type=file]').attachFile('@responseBody')
+```
+
+### Processing data inside the test
+
+```javascript
+cy.readFile('users.json').then((users => {
+  users[0].username = 'JohnCena'
+})).as('myFile')
+
+cy.get('input[type=file]').attachFile('@myFile')
+```
+
+### Attaching multiple files
+
+```javascript
+cy.get('input[type=file]').attachFile(['file1.json', 'file2.json', 'file3.json'])
+```
+
+This will fail unless the file input has the `multiple` property.
+
+### Attaching a file with custom filename and lastModified
+
+```javascript
+cy.get('input[type=file]').attachFile({
+  contents: 'path/to/file.json',
+  fileName: 'users.json',
+  lastModified: new Date('Feb 18 1989').valueOf()
+})
+```
+
+### Dropping a file on the document
+
+```javascript
+cy.document().attachFile('file.json', { action: 'drag-n-drop' })
+```
+
+## Notes
+
+### Existence
+
+#### Default file existence assertion
+
+When passed a file path, `cy.attachFile()` asserts that the file exists and will fail if it does not exist. It will retry reading the file if it does not initially exist until the file exists or the command times out.
```

Comment:
```suggestion
When passed a file path, `.attachFile()` asserts that the file exists and will fail if it does not exist. It will retry reading the file if it does not initially exist until the file exists or the command times out.
```